### PR TITLE
[staging] pythonPackages.pyyaml: 5.3.1 -> 5.4.1

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg build ceph

**Comment by dotlambda** [comment]
@ofborg build ceph

**Comment by jonringer** [comment]
@GrahamcOfBorg build awscli

**Comment by jonringer** [comment]
seems ceph is broken for other reasons

**Comment by dotlambda** [comment]
This fixes CVE-2020-14343: https://github.com/yaml/pyyaml/blob/5.4.1/CHANGES

**Comment by dotlambda** [comment]
@ofborg test awscli

**Comment by dotlambda** [comment]
I successfully ran the awscli test on my machine and built some other random packages.
However, I noticed that ddt fails: https://github.com/datadriventests/ddt/issues/95.

**Comment by dotlambda** [comment]
I suggest we merge and fix stuff later if we notice major breakage on Hydra.

File: pkgs/servers/home-assistant/default.nix [hunk 1]
```diff
@@ -84,11 +84,11 @@ in with py.pkgs; buildPythonApplication rec {
 
   postPatch = ''
     substituteInPlace setup.py \
-      --replace "aiohttp==3.7.1" "aiohttp>=3.6.3" \
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/default.nix:87]: That is not unpinning PyYAML.

Comment (dotlambda) [reply pkgs/servers/home-assistant/default.nix:87]: Done

**Review by mweinelt — APPROVED** [review]
```
/nix/store/ylpi5d68cx7rzbj462rizk5vhnilyb44-python3.8-PyYAML-5.4.1
/nix/store/dpc8948ssipm5rh3np4vrr9ymcambzw8-python3.7-PyYAML-5.4.1
/nix/store/8rn7rffvmnap2czlghb66yincy2s8vn7-python3.9-PyYAML-5.4.1
```

**Comment by dotlambda** [comment]
20.09: https://github.com/NixOS/nixpkgs/pull/110525

**Pull request merged**